### PR TITLE
Clarify that MSVC build shell requirements

### PR DESCRIPTION
**Pull request by nikitalita**
Add a note that MSVC builds can't be done from the MSYS2 or MINGW* shells, and that cmd.exe or PowerShell should be used instead.

### CODE REVIEW

**Comment by nikitalita** [comment]
Is this what you intended?

File: development/compiling/compiling_for_windows.rst [hunk 1]
```diff
@@ -100,13 +100,16 @@ Selecting a compiler
...
+use MinGW, pass ``use_mingw=yes`` to the SCons command line. Note that MSVC 
+builds cannot be performed from the MSYS2 or MingW* shells, use either 
+`cmd.exe` or PowerShell instead.
 
 During development, using the Visual Studio compiler is usually a better idea,
 as it links the Godot binary much faster than MinGW. However, MinGW can
 produce more optimized binaries using link-time optimization (see below),
 making it a better choice for production use.
 
+
```
Comment (Calinou) [inline development/compiling/compiling_for_windows.rst:112]:
```suggestion

```

File: development/compiling/compiling_for_windows.rst [hunk 2]
```diff
@@ -100,13 +100,14 @@ Selecting a compiler
...
 MinGW instead. If you already have Visual Studio installed and want to
-use MinGW, pass ``use_mingw=yes`` to the SCons command line.
+use MinGW, pass ``use_mingw=yes`` to the SCons command line. Note that MSVC 
+builds cannot be performed from the MSYS2 or MingW* shells, use either 
+`cmd.exe` or PowerShell instead.
 
 During development, using the Visual Studio compiler is usually a better idea,
 as it links the Godot binary much faster than MinGW. However, MinGW can
 produce more optimized binaries using link-time optimization (see below),
 making it a better choice for production use.
```
Comment (Calinou) [inline development/compiling/compiling_for_windows.rst:110]:
```suggestion
produce more optimized binaries using link-time optimization (see below),
making it a better choice for production use.

```

File: development/compiling/compiling_for_windows.rst [hunk 3]
```diff
@@ -100,7 +100,9 @@ Selecting a compiler
 SCons will automatically find and use an existing Visual Studio installation.
 If you do not have Visual Studio installed, it will attempt to use
 MinGW instead. If you already have Visual Studio installed and want to
-use MinGW, pass ``use_mingw=yes`` to the SCons command line.
+use MinGW, pass ``use_mingw=yes`` to the SCons command line. Note that MSVC 
+builds cannot be performed from the MSYS2 or MingW* shells, use either 
+`cmd.exe` or PowerShell instead.
```
Comment (Calinou) [inline development/compiling/compiling_for_windows.rst:105]:
```suggestion
use MinGW, pass ``use_mingw=yes`` to the SCons command line. Note that MSVC 
builds cannot be performed from the MSYS2 or MinGW shells. Use either 
`cmd.exe` or PowerShell instead.
```

**Review by Calinou — APPROVED** [review]
Thanks!

**Pull request merged**